src/lib/operators/join_sort_merge.cpp

```diff
@@ -517,6 +732,34 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
 
   // Performs the join on a single cluster. Runs of entries with the same value are identified and handled together.
   // This constitutes the merge phase of the join. The output combinations of row ids are determined by _join_runs.
+  // The main logic of the algorithm is the following:
+  // The values of the left and right join columns are grouped into runs (entries with the same value). The values of
+  // the runs are compared and then we determine if we advanced the right run, left run or both runs. If the comparison
+  // result is equal, we advance both runs, if it is less we advance the left one if it is greater we advance the right
```

Comment:
"equal, we advance both runs. If it is less, we advance the left one. If it is greater, we ...".